Apps/Src/ReadCarCAN.c

```diff
@@ -5,47 +5,49 @@
 #include "Contactors.h"
 #include "Minions.h"
 #include "CAN_Queue.h"
+#include "FaultState.h"
+#include "os_cfg_app.h"
 
+// watchDogTripCounter variable and limit
+static int CANWatchTripCounter = 0; 
+static const int CANWATCH_TRIP_LIMIT = 2;
 
-static bool msg_recieved = false;
-static OS_MUTEX msg_rcv_mutex;
 
+//MMM CAN watchdog timer variables
+static OS_TMR canWatchTimer; // Is it ok to declare this here?
+static const int CAN_WATCH_TMR_DLY_SECONDS = 5;
 
-static int watchDogTripCounter = 0; //count how many times the CAN watchdog trips
+//MMM prechargeDlyTimer
+static OS_TMR prechargeDlyTimer;
 
-//CAN watchdog thread variables
-static OS_TCB canWatchTCB;
-static CPU_STK canWatchSTK[DEFAULT_STACK_SIZE];
+//MMM restart array and disable charging semaphonres
+static OS_SEM restartArraySem;
+static OS_SEM canWatchDisableChargingSem;
 
-//Array restart thread variables
-static OS_TCB arrayTCB;
-static CPU_STK arraySTK[DEFAULT_STACK_SIZE];
 
 // Array restart thread lock
 static OS_MUTEX arrayRestartMutex;
 static bool restartingArray = false;
 
-static void CANWatchdog_Handler(); //Handler if we stop getting messages
-static void ArrayRestart(); //handler to turn array back on
+/*
+static void CANWatchdog_Handler(); //Handler if we stop getting messages */
+static void arrayRestart(); //handler to turn array back on
```

Comment:
You should probably put the arguments in this forward declaration. This is technically correct, but looks confusing.